src/Service/MagicLinks/MagicLinkService.cs

```diff
@@ -0,0 +1,49 @@
+using Passwordless.Common.Services.Mail;
+using Passwordless.Service.MagicLinks.Models;
+using Passwordless.Service.Models;
+
+namespace Passwordless.Service.MagicLinks;
+
+public class MagicLinkService
+{
+    private readonly IFido2Service _fido2Service;
+    private readonly IMailProvider _mailProvider;
+
+    public MagicLinkService(IFido2Service fido2Service, IMailProvider mailProvider)
+    {
+        _fido2Service = fido2Service;
+        _mailProvider = mailProvider;
+    }
+
+    public async Task SendMagicLink(MagicLinkDTO dto)
+    {
+        var token = await _fido2Service.CreateSigninToken(new SigninTokenRequest(dto.UserId));
+
+        var link = new Uri(dto.LinkTemplate.Replace("<token>", token));
+
+        await _mailProvider.SendAsync(new MailMessage
+        {
+            To = [dto.EmailAddress.ToString()],
+            From = "do-not-reply@passwordless.dev",
+            Subject = "Verify Email Address",
```

Comment:
```suggestion
	    MessageStream = "free-magic-links",
            Subject = "Verify Email Address",
```

Untested, I hope the SDK method matches with the RestAPI.